_source/_posts/2019-11-04-build-net-app-quickly-with-docker.md

```diff
@@ -0,0 +1,225 @@
+
+```xml
+<add key="okta:ClientId" value="{clientId}" />
+<add key="okta:ClientSecret" value="{clientSecret}" />
+<add key="okta:OktaDomain" value="{yourOktaDomain}" />
+```
+
+If you don't already have an Okta account, you can sign up for a [free-forever developer account](https://developer.okta.com/signup). Once logged into your Okta account, you'll see your Okta domain at the top right of your developer dashboard.
+
+{% img blog/docker-net-fx/okta-domain-url.png alt:"Your Okta Domain URL" width:"800" %}{: .center-image }
+
+To create a new application, select **Applications** from the main menu and click the **Add Application** button at the top of the applications list. Then choose **Web Application** and click **Next**.
+
+{% img blog/docker-net-fx/okta-new-application.png alt:"New Okta Application" width:"800" %}{: .center-image }
+
+To configure the application, set the values to look like below, and then click **Done**.
+
+{% img blog/docker-net-fx/okta-app-settings.png alt:"Okta Application Settingss" width:"800" %}{: .center-image }
+
+At the bottom of the **General Settings** tab, you'll see a section with **Client ID** and **Client Secret**. Copy those into the corresponding configuration values from above.
+
+Once you've added the correct configuration values, you'll be able to launch the application using **Ctrl-F5** or **Debug > Start Without Debugging**. Point your browser at <https://localhost:44300> and add some Launches (after authenticating, of course!) to make sure everything is working correctly.
+
+## Docker and the .Net Framework
+
+Now you've got a working sample application, you can containerize it with Docker for Windows. The process is pretty much the same for Windows containers as for Linux containers.
+
+### Add a Dockerfile to Your .NET Application
+
+Visual Studio provides some useful tooling to containerize applications. To easily set up your application to use Docker and Docker Compose,right-click on the project node (`CrudMVCCodeFirst`) and then select **Add > Container Orchestrator Support**. In the dialog that appears, select **Docker Compose**. Then, as if by magic, Visual Studio adds a `Dockerfile` and `docker-compose` file to your application and things start building in the background.
+
+Once the build process completes, start the application and  keep an eye on the URL. You'll see that it's pointing to <http://172.x.x.x> What's going on?
+
+> __Note__: It can take a long time to build a Docker image the first time around. The build process must first download the appropriate base image, which are generally large for Windows Containers. If you're not blessed with a Gigabit internet connection, have some coffee, and don't worry.We'll wait!
```

Comment:
```suggestion
> __Note__: It can take a long time to build a Docker image the first time around. The build process must first download the appropriate base image, which is generally large for Windows Containers. If you're not blessed with a Gigabit internet connection, have some coffee, and don't worry. We'll wait!
```